src/snowflake/cli/api/utils/graph.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Generic, Optional, TypeVar
+
+from click import ClickException
+
+T = TypeVar("T")
+
+
+@dataclass
+class Node(Generic[T]):
+    key: str
+    data: T
+    neighbors: set[Node[T]] = field(default_factory=set)
+    status: Optional[str] = None
```

Comment:
Suggest that you use an Enum instead of a raw string, and instead of None being the default have it be "UNVISITED".